lib/karafka/active_job/consumer.rb

```diff
@@ -24,21 +24,26 @@ def consume
       #
       # @param job_message [Karafka::Messages::Message] message with active job
       def consume_job(job_message)
-        # We technically speaking could set this as deserializer and reference it from the
-        # message instead of using the `#raw_payload`. This is not done on purpose to simplify
-        # the ActiveJob setup here
-        job = ::ActiveSupport::JSON.decode(job_message.raw_payload)
+        with_deserialized_job(job_message) do |job|
+          tags.add(:job_class, job['job_class'])
 
-        tags.add(:job_class, job['job_class'])
+          payload = { caller: self, job: job, message: job_message }
 
-        payload = { caller: self, job: job, message: job_message }
-
-        # We publish both to make it consistent with `consumer.x` events
-        Karafka.monitor.instrument('active_job.consume', payload)
-        Karafka.monitor.instrument('active_job.consumed', payload) do
-          ::ActiveJob::Base.execute(job)
+          # We publish both to make it consistent with `consumer.x` events
+          Karafka.monitor.instrument('active_job.consume', payload)
+          Karafka.monitor.instrument('active_job.consumed', payload) do
+            ::ActiveJob::Base.execute(job)
+          end
         end
       end
+
+      # @param job_message [Karafka::Messages::Message] message with active job
+      def with_deserialized_job(job_message)
```

Comment:
Ideally, if we had a middleware system like Rack or Sidekiq, we could hook the CurrentAttributes code into that. 
For the current approach, I created two methods in the Dispatcher and Consumer so that we have a way to hook CurrentAttributes into the job's serialization/deserialization process.